actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -394,9 +428,11 @@ def setup(context, options, block)
         @variable, @variable_counter, @variable_iteration = retrieve_variable(@path, as)
         @template_keys = retrieve_template_keys
       else
-        paths.map! { |path| retrieve_variable(path, as).unshift(path) }
+        @collection_data.map! { |path| retrieve_variable(path, as).unshift(path) }
```

Comment:
Notice how `@collection_data` is assigned to `paths` on line 415. Found that confusing as hell, so I'd rather use the ivar.